testing/run_tests.py

```diff
@@ -0,0 +1,229 @@
+
+  assert os.path.exists(dart)
+  assert os.path.exists(frontend_server)
+  assert os.path.exists(flutter_patched_sdk)
+  assert os.path.exists(test_packages)
+  
+  snapshot_command = [
+    dart,
+    frontend_server,
+    '--sdk-root',
+    flutter_patched_sdk,
+    '--incremental',
+    '--strong',
+    '--target=flutter',
+    '--packages',
+    test_packages,
+    '--output-dill',
+    kernel_file_output,
+    dart_file
+  ]
+  
+  subprocess.check_call(snapshot_command, cwd=buildroot_dir)
+  assert os.path.exists(kernel_file_output)
+
+
+def RunDartTest(build_dir, dart_file, filter):
+  kernel_file_name = os.path.basename(dart_file) + '.kernel.dill'
+  kernel_file_output = os.path.join(out_dir, kernel_file_name)
+  
+  SnapshotTest(build_dir, dart_file, kernel_file_output)
+
+  print "Running test '%s' using 'flutter_tester'" % kernel_file_name
+  RunEngineExecutable(build_dir, 'flutter_tester', filter, [
+    '--disable-observatory',
+    '--use-test-fonts',
+    kernel_file_output
+  ])
+
+def RunPubGet(build_dir, directory):
+  print "Running 'pub get' in the tests directory %s" % dart_tests_dir
+
+  pub_get_command = [
+    os.path.join(build_dir, 'dart-sdk', 'bin', 'pub'),
+    'get'
+  ]
+  subprocess.check_call(pub_get_command, cwd=directory)
+
+
+def EnsureDebugUnoptSkyPackagesAreBuilt():
+  variant_out_dir = os.path.join(out_dir, 'host_debug_unopt')
+
+  ninja_command = [
+    'autoninja',
+    '-C',
+    variant_out_dir,
+    'flutter/sky/packages'
+  ]
+
+  # Attempt running Ninja if the out directory exists.
+  # We don't want to blow away any custom GN args the caller may have already set.
+  if os.path.exists(variant_out_dir):
+    subprocess.check_call(ninja_command, cwd=buildroot_dir)
+    return
+
+  gn_command = [
+    os.path.join(buildroot_dir, 'flutter', 'tools', 'gn'),
+    '--runtime-mode',
+    'debug',
+    '--unopt',
+    '--no-lto',
+  ]
+  
+  subprocess.check_call(gn_command, cwd=buildroot_dir)
+  subprocess.check_call(ninja_command, cwd=buildroot_dir)
+
+def RunDartTests(build_dir, filter):
+  # This one is a bit messy. The pubspec.yaml at flutter/testing/dart/pubspec.yaml
+  # has dependencies that are hardcoded to point to the sky packages at host_debug_unopt/
+  # Before running Dart tests, make sure to run just that target (NOT the whole engine)
+  EnsureDebugUnoptSkyPackagesAreBuilt();
+
+  # Now that we have the Sky packages at the hardcoded location, run `pub get`.
+  RunPubGet(build_dir, dart_tests_dir)
+
+  dart_tests = glob.glob('%s/*.dart' % dart_tests_dir)
+
+  for dart_test_file in dart_tests:
+    if filter in os.path.basename(dart_test_file):
+      print "Testing dart file %s" % dart_test_file
+      RunDartTest(build_dir, dart_test_file, filter)
+    else:
+      print "Skipping %s due to filter." % dart_test_file
+
+
+def RunTests(build_dir, filter, run_engine_tests, run_dart_tests, run_benchmarks):
+  if run_engine_tests:
+    RunEngineTests(build_dir, filter)
```

Comment:
Nit: the dart tests seem to be also engine tests. Shall we call this `RunGtests`, `RunCppTests`, `RunCcTests`, or just `RunUnitTests`?

Reply:
Done. Used `RunCCTests`